### ISSUE TITLE
Add `react` and `react-dom` to externals

### ISSUE THREAD

**Issue by brendo** [mac-s-g/react-json-view#75]
While debugging my projects bundle size, this module came in as the largest dependency.

I had a look and noticed that `react` and `react-dom` are included in the bundle for rjv:

```
$ webpack --json | webpack-bundle-size-analyzer
react-dom: 538.5 KB (45.0%)
react: 82.47 KB (6.88%)
radium: 63.25 KB (5.28%)
inline-style-prefixer: 56.87 KB (4.75%)
react-tooltip: 49.97 KB (4.17%)
core-js: 36.45 KB (3.04%)
lodash.curry: 35.03 KB (2.92%)
fbjs: 30.79 KB (2.57%)
create-react-class: 27.71 KB (2.31%)
prop-types: 23.38 KB (1.95%)
base16: 22.03 KB (1.84%)
bowser: 17.29 KB (1.44%)
clipboard: 15.31 KB (1.28%)
lodash.flow: 11.37 KB (0.949%)
react-icons: 10.49 KB (0.875%)
react-base16-styling: 9.73 KB (0.812%)
events: 8.13 KB (0.679%)
flux: 7.46 KB (0.623%)
process: 5.29 KB (0.442%)
pure-color: 3.45 KB (0.288%)
good-listener: 3.37 KB (0.282%)
babel-runtime: 3.29 KB (0.275%)
object-assign: 2.06 KB (0.172%)
react-icon-base: 1.95 KB (0.163%)
delegate: 1.82 KB (0.152%)
tiny-emitter: 1.53 KB (0.128%)
classnames: 1.08 KB (0.0898%)
select: 1 KB (0.0836%)
exenv: 863 B (0.0703%)
webpack: 597 B (0.0487%)
hyphenate-style-name: 339 B (0.0276%)
<self>: 125.14 KB (10.4%)

Note: The file sizes are calculated before minification, and might not reflect the real file sizes in the production bundle.
Done in 10.84s.
```

I feel like they could instead added as externals, as it'd be implied that if someone wanted to use this component, that they should be including React/ReactDOM themselves:

```
  externals: {
    'cheerio': 'window',
    'react': 'React',
    'react-dom': 'ReactDom',
    'react/lib/ExecutionEnvironment': true,
    'react/lib/ReactContext': true,
  },
```

After analysis:

```
$ webpack --json | webpack-bundle-size-analyzer
radium: 63.25 KB (12.3%)
inline-style-prefixer: 56.87 KB (11.0%)
react-tooltip: 49.97 KB (9.70%)
core-js: 36.45 KB (7.08%)
lodash.curry: 35.03 KB (6.80%)
prop-types: 22.51 KB (4.37%)
base16: 22.03 KB (4.28%)
bowser: 17.29 KB (3.36%)
clipboard: 15.31 KB (2.97%)
lodash.flow: 11.37 KB (2.21%)
react-icons: 10.49 KB (2.04%)
react-base16-styling: 9.73 KB (1.89%)
events: 8.13 KB (1.58%)
flux: 7.46 KB (1.45%)
fbjs: 4.71 KB (0.914%)
pure-color: 3.45 KB (0.670%)
good-listener: 3.37 KB (0.655%)
babel-runtime: 3.29 KB (0.639%)
react-icon-base: 1.95 KB (0.379%)
delegate: 1.82 KB (0.354%)
tiny-emitter: 1.53 KB (0.297%)
classnames: 1.08 KB (0.209%)
select: 1 KB (0.195%)
exenv: 863 B (0.164%)
webpack: 597 B (0.113%)
hyphenate-style-name: 339 B (0.0643%)
<self>: 125.23 KB (24.3%)

Note: The file sizes are calculated before minification, and might not reflect the real file sizes in the production bundle.
Done in 7.34s.
```

What do you think?

**Comment by mac-s-g** [comment]
wow.  i'm so glad you pointed this out. 
this is really helpful,  i'm updating dependency mgmt right now.  i'll update when i post a PR.

**Comment by mac-s-g** [comment]
posted #76 

Bundle size looks quite a bit better.

I could continue to improve bundle size by: 
* replacing react-tooltip dependency with lightweight built-in pop up
* investigating size of babel-runtime dependency
* investigating size of inline-style-prefixer
* investigating size of all the base-16 style stuff

**Comment by mac-s-g** [comment]
published react-json-view@1.10.4

it's down to about 230kb total.

still pretty large, but quite a bit better

**Comment by brendo** [comment]
Thanks!
